core/src/sumeragi/main_loop.rs

```diff
@@ -881,57 +859,63 @@ pub(crate) fn run(
             &mut view_change_time,
         );
 
+        if let Some(message) = {
+            let (msg, sleep) = sumeragi.receive_network_packet(&mut view_change_proof_chain);
+            should_sleep = sleep;
+            msg
+        } {
+            handle_message(
+                message,
+                &mut sumeragi,
+                &mut voting_block,
+                current_view_change_index,
+                &mut voting_signatures,
+            );
+        }
+
+        // State could be changed after handling message so it is necessary to reset state before handling message independent step
```

Comment:
You only partially moved reset state part, did you do this on purpose?

Reply:
Yes.